### DISCUSSION TITLE
Вернуть список только уникальных значений

### DISCUSSION THREAD

**Discussion by gorbee** [totumonline/totum-ru-issues-and-discussions#503] [Как это сделать?] [answered]
Привет! Есть прекрасная функция `listUniq`, но она возвращает уникальные значения, даже если это значение было неуникальным 

**Из документации:**
```
example: listUniq(list: $list)
list: listCreate(item: "Alex"; item: "Sam"; item: "Alex")

// Результат: ["Alex","Sam"]
```

Моя задача состоит в том, чтобы вернуть список уникальных значений, но если это значение неуникально, то его нужно удалить.

То-есть, если взять пример из документации, по моей задаче код должен вернуть только `["Sam"]`, так как "Alex" - неуникальное значение в списке.

Несколько дней пытался решить разными способами, смотрел дискуссии, но безуспешно. Возможно, плохо искал. Подскажите, пожалуйста, как такое можно сделать?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
```
= : $filter[[value]]

filter: listFilter(list: $replace; key: "count" = 1)

replace: listReplace(list: $rowlist; action: "count" = $count; value: "replace")
		rowlist: rowListCreate(field: "value" = $list)
			~list: listCreate(item: "Alex"; item: "Sam"; item: "Alex"; item: "John")
	count: listCount(list: $inner_filter)
		inner_filter: listFilter(list: $list; key: "value" = $#replace[value])
```

Сегодня нет времени объяснять че тут происходит, но он решает вашу задачу.
